packages/glimmer-runtime/lib/compiled/opcodes/content.ts

```diff
@@ -45,33 +65,6 @@ export class AppendOpcode extends Opcode {
   }
 }
 
-export class UpdateAppendOpcode extends UpdatingContentOpcode {
-  type = 'update-append';
-  private cache: ReferenceCache<string>;
-  private textNode: Text;
-
-  constructor(cache: ReferenceCache<string>, textNode: Text) {
-    super();
-    this.cache = cache;
-    this.textNode = textNode;
-  }
-
-  evaluate() {
-    let value = this.cache.revalidate();
-    if (isModified(value)) this.textNode.nodeValue = value;
-  }
-
-  toJSON(): OpcodeJSON {
-    let { _guid: guid, type } = this;
-
-    let details = dict<string>();
-
-    details["lastValue"] = JSON.stringify(this.cache.peek());
-
-    return { guid, type, details };
-  }
-}
-
 export class TrustingAppendOpcode extends Opcode {
```

Comment:
I think `TrustingAppendOpcode` and the corresponding updating opcode also needs to handle `SafeString`s. The difference is you will *always* insert things as HTML.. i.e. `insertHTMLBefore( isSafeString(value) ? valu.toHTML() : value )`